### PR TITLE
Adding the ability to set provenance hash

### PR DESCRIPTION
**Pull request by Shiva-Tzu**
The provenance hash is an important feature for establishing trust between customer and seller if a reveal is occurring. Noticed it was not in the contract so decided to add the functionality in both contract and package.json commands.

### CODE REVIEW

**Comment by jeongmincho** [comment]
nice!!! thanks:) i'll review this by EOW

File: smart-contracts/package.json [hunk 1]
```diff
@@ -21,6 +21,7 @@
     "giftTokens": "npx hardhat run scripts/giftTokens.js",
     "reserveTokens": "npx hardhat run scripts/reserveTokens.js",
     "setBaseURI": "npx hardhat run scripts/setBaseURI.js",
+     "setProvenanceHash": "npx hardhat run scripts/setProvenanceHash.js"
```
Comment (jeongmincho) [inline smart-contracts/package.json:24]:
```suggestion
    "setProvenanceHash": "npx hardhat run scripts/setProvenanceHash.js",
```

File: smart-contracts/scripts/setProvenanceHash.js [hunk 2]
```diff
@@ -0,0 +1,27 @@
+require("dotenv").config();
+const hre = require("hardhat");
+
+async function main() {
+  const PROVENANCE_HASH = process.env.PROVENANCE_HASH;
+  if (!PROVENANCE_HASH) {
+    console.log("BASE_URI is required. Please add it to your environment.");
```
Comment (jeongmincho) [inline smart-contracts/scripts/setProvenanceHash.js:7]:
```suggestion
    console.log("PROVENANCE_HASH is required. Please add it to your environment.");
```

File: smart-contracts/scripts/setProvenanceHash.js [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+    return;
+  }
+
+  const TechnoFeudal = await hre.ethers.getContractFactory("TechnoFeudal");
+  const nft = await TechnoFeudal.attach(
+    process.env.CONTRACT_ADDRESS // The deployed contract address
+  );
+  console.log("TechnoFeudal attached to:", nft.address);
```
Comment (jeongmincho) [inline smart-contracts/scripts/setProvenanceHash.js:15]:
```suggestion
  const MyNFT = await hre.ethers.getContractFactory("MyNFT");
  const nft = await MyNFT.attach(
    process.env.CONTRACT_ADDRESS // The deployed contract address
  );
  console.log("MyNFT attached to:", nft.address);
```

**Review by jeongmincho — CHANGES_REQUESTED** [review]
hey @Shiva-Tzu! sorry for the delay. thanks again for adding this! provenance hashes are important for sure. added a few stylistic suggestions but overall looks great. let's get you merged in asap.

**Comment by Shiva-Tzu** [comment]
Great! Glad I could assist!

**Pull request merged**